server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -438,7 +439,7 @@ protected void onShardGroupFailure(int shardIndex, SearchShardTarget shardTarget
     @Override
     public final void onShardFailure(final int shardIndex, @Nullable SearchShardTarget shardTarget, Exception e) {
         // we don't aggregate shard failures on non active shards (but do keep the header counts right)
-        if (TransportActions.isShardNotAvailableException(e) == false) {
+        if (TransportActions.isShardNotAvailableException(e) == false && (requestCancelled.get() && isTaskCancelledException(e)) == false) {
```

Comment:
Can you update the comment above to say that we don't aggregate shard failures on search cancelled internally ?

Reply:
++, added in https://github.com/elastic/elasticsearch/pull/64240/commits/cd53c74a057a1bcddfe7d7429643607f993312cf